### ISSUE TITLE
using multiprocess context (for dill support)  no longer working

### ISSUE THREAD

**Issue by sergeykk** [noxdafox/pebble#128]
listed in pebble documentation code (with multiprocess context) is not working, details:
- documentation: https://pebble.readthedocs.io/en/latest/
- python 3.11.6
- error:
```
/apps/quorumfs1/skurganov/dev/monorepo/core/research/task_graph/example2.py
Traceback (most recent call last):
  File "/apps/quorumfs1/skurganov/dev/monorepo/core/research/task_graph/example2.py", line 7, in <module>
    @concurrent.process(context=multiprocess.get_context("spawn"))
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/n/local2/condacem/.micromamba/quorum/b1248e675859d49c0b164499688be54f/lib/python3.11/site-packages/pebble/concurrent/process.py", line 65, in process
    _validate_parameters(timeout, name, daemon, mp_context)
  File "/n/local2/condacem/.micromamba/quorum/b1248e675859d49c0b164499688be54f/lib/python3.11/site-packages/pebble/concurrent/process.py", line 199, in _validate_parameters
    raise TypeError('Context expected to be None or multiprocessing.context')
TypeError: Context expected to be None or multiprocessing.context
```
- code: 
```
import threading
import multiprocess

from pebble import concurrent


@concurrent.process(context=multiprocess.get_context("spawn"))
def function(unpicklable_object):
    """Pickle an unpicklable object."""
    unpicklable_object.pickle_this()


class Unpicklable:
    """This class cannot be serialized by standard Python Pickle."""

    def __init__(self):
        self.lock = threading.Lock()

    def pickle_this(self):
        with self.lock:
            print("Just pickled!")


def main():
    unpicklable = Unpicklable()
    future = function(unpicklable)
    future.result()


if __name__ == "__main__":
    main()

```

**Comment by noxdafox** [comment]
This was introduced in https://github.com/noxdafox/pebble/commit/36eefa497dd8f5596bcb9606f61de556b4c15939. 

Will fix it ASAP, thanks!

**Comment by noxdafox** [comment]
Issue fixed in release `5.0.7`, please re-open this issue if the problem persists.
